frame/collective/src/lib.rs

```diff
@@ -666,12 +566,17 @@ decl_module! {
 		///  - any mutations done while executing `proposal` (`P1`)
 		/// - up to 3 events
 		/// # </weight>
+		// NOTE: surprising difference between old manual weight and new generated weight:
+		// early disapproved: 37.21 vs 47.412
+		// early approved:    50.82 vs 69.466 and B * 0.008 vs 0.004 * b
+		// disapproved:       B * 0.003 vs 0 * b
+		// approved:          65.95 vs 77.033
 		#[weight = (
 			weight_for::close::<T, I>(
-				MAX_MEMBERS.into(), // `M`
+				*length_bound, // B
+				MAX_MEMBERS, // `M`
 				*proposal_weight_bound, // `P1`
-				T::MaxProposals::get().into(), // `P2`
-				*length_bound as Weight, // B
+				T::MaxProposals::get(), // `P2`
 			),
```

Comment:
actually I would be in favor of writing everything here, as you don't reuse close function anyway:
```rust
		#[weight = (
			{
				let b = *length_bound;
				let m = MAX_MEMBERS;
				let p1 = *proposal_weight_bound;
				let p2 = T::MaxProposals::get();
				T::WeightInfo::close_early_approved(b, m, p2)
					.max(T::WeightInfo::close_early_disapproved(m, p2))
					.max(T::WeightInfo::close_approved(b, m, p2))
					.max(T::WeightInfo::close_disapproved(m, p2))
					.saturating_add(p1)
			},
			DispatchClass::Operational
		)]
```
But I'm ok anyway